### ISSUE TITLE
gem install doesn't update .so files if changed

### ISSUE THREAD

**Issue by wlevine** [rubygems/rubygems#1285]
I'm trying to create a gem which includes a C extension. I'm using `Gem::PackageTask` to make the gem.

When I install the gem it builds the .so file properly and installs it in three locations:

```
/home/me/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/extensions/x86_64-linux/2.2.0-static/nmatrix-atlas-0.1.0/nmatrix_atlas.so
/home/me/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/nmatrix-atlas-0.1.0/lib/nmatrix_atlas.so
/home/me/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/nmatrix-atlas-0.1.0/ext/nmatrix_atlas/nmatrix_atlas.so
```

The first one is the one that actually gets loaded when I do `require 'whatever'`. The problem is that when I make a change to the code, then repackage the gem, then run `gem install` again, the new code gets compiled into a new .so and the latter two .so's get replaced with the new one, but the first one doesn't get updated. So ruby still sees the old code. I have to do `gem uninstall` followed by `gem install` to get it to update.

Is this a bug in rubygems? Or is it the expected behavior? Someone on stack overflow suggested that it wasn't overwriting the file because I hadn't bumped the version. But then why did it overwrite all the other files. Am I doing something else wrong?

**Comment by copiousfreetime** [comment]
This is a duplicate of #1192, and there is a fix pending in #1576

**Comment by bronzdoc** [comment]
This was fixed by #1576
